src/Controls/src/Core/Handlers/Shell/ShellItemHandler.Windows.cs

```diff
@@ -169,9 +169,15 @@ private void OnNavigationTabChanged(NavigationView sender, NavigationViewSelecti
 
 				((Shell)VirtualView.Parent).CurrentItem = shellSection;
 			}
-			else if (selectedItem.Data is ShellContent shellContent)
+			else if (selectedItem.Data is ShellContent shellContent && VirtualView.Parent is Shell parentShell)
 			{
-				((Shell)VirtualView.Parent).CurrentItem = shellContent;
+				// We need to invoke ProposeSection for TabBar items navigation for ShellContent
+				var currentItem = parentShell.CurrentItem?.CurrentItem;
+				if (currentItem?.Title != shellContent.Title && currentItem != shellContent.Parent)
+				{
+					(parentShell.CurrentItem as IShellItemController)?.ProposeSection(shellContent);
```

Comment:
Is there anywhere else we can call this so that the user can cancel the navigation? 

The point of ProposeSection is to give the user time to cancel the navigation